falco.yaml

```
@@ -773,6 +775,39 @@ metrics:
   convert_memory_to_mb: true
   include_empty_values: false
 
+  ###############################################
+# Falco driver selection                    #
+###############################################
+
+# [Stable] `driver_mode`
+#
+# --- [Description]
+#
+# Falco supports different driver modes for capturing syscall events. The choice
+# of driver mode can significantly impact the performance and compatibility of
+# Falco with your system. Choose the appropriate driver mode based on your
+# system's configuration and requirements.
+#
+# Available driver modes:
+# - `kmod`: Kernel Module (Kernel Module)
+# - `bpf`: eBPF (Extended Berkeley Packet Filter)
+# - `modern_bpf`: Modern eBPF (Modern Extended Berkeley Packet Filter)
+# - `nodriver`: No Driver (No driver, just for testing)
+# - `gvisor`: gVisor (gVisor sandbox)
+# - `custom`: Custom Driver (Specify a custom driver module)
+
+# Example usage:
+# driver_mode: kmod
+
+# Select the appropriate driver mode by uncommenting the corresponding line.
+# Make sure to specify only one driver mode at a time.
+
+driver_mode: kmod
```

```suggestion
driver: 
   kind: kmod
```

I'd prefer to have the ability to include suboptions to `driver` to avoid polluting the config with first-level keys. For example:

```yaml
driver:
  kind: ebpf
  ebpf:
    # Path where the eBPF probe is located. 
    path: /root/.falco/
  modern_ebpf:
    cpus_for_each_syscall_buffer: 2
```

This will mimic [the Falco Helm chart config](https://github.com/falcosecurity/charts/blob/master/charts/falco/values.yaml#L177), which is fine IMO
 
**N.B. re. naming convention**
I believe we als have to fix the naming convention of this stuff for the last time. My proposal is as follows and include some alias to avoid headaches:

- `nodriver` (alias: `no-driver`)
- `module` (alias: `kmod`)
- `ebpf` (alias: `bpf`)
- `modern-ebpf`: (alias: `modern-bpf`)
- `gvisor` (which is technically not a driver, but I don't think we have a better option atm)

---

i tend to agree with this design and naming, we should follow the same approach for helm charts and systemd units

---

I really like it! I'll update it!

---

when we decide something definitive, we should update this issue https://github.com/falcosecurity/falco/issues/2840: "deprecation" in Falco 0.37 means "breaking change" in Falco 0.38, so I would directly update this issue

---

I agree with the design; about the naming, i'd go with:
* `nodriver` alias `no-driver`
* `module` alias `kmod`
* `bpf`
* `modern-bpf`
* `gvisor`

We never used `ebpf` nor `modern-ebpf` anywhere afaik. We have `FALCO_BPF_PROBE` and `--modern-bpf` today in Falco, and i'd keep them.

Falcoctl will use: `kmod`, `bpf`, `modern-bpf`.
Once again remember the naming of our systemd services too (https://github.com/falcosecurity/falco/tree/master/scripts/systemd)!

---

we `ebpf` in the helm chart and we tend to use `ebpf` instead of `bpf` in all the documentation https://falco.org/tags/ebpf/ because technically it is more appropriate... but I have no strong opinion here we just need to pick one

---

Oh i see; thanks for pointing that out!

---

I'm ok with picking `ebpf` as the standard one. Also, I don't see any issue with maintaining aliases.